app/post.go

```diff
@@ -298,6 +293,40 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 	return rpost, nil
 }
 
+func (a *App) attachFilesToPost(post *model.Post) *model.AppError {
+	result := <-a.Srv.Store.FileInfo().AttachMultipleToPost(post.FileIds, post.Id, post.UserId)
+	if result.Err != nil {
+		return result.Err
+	}
+
+	count := result.Data.(int64)
+	if count != -1 && count != int64(len(post.FileIds)) {
+		// We couldn't attach all files to the post, so ensure that post.FileIds reflects what was actually attached
+		mlog.Warn("Failed to attach all files to the post", mlog.String("post_id", post.Id))
+
+		result := <-a.Srv.Store.FileInfo().GetForPost(post.Id, true, false)
```

Comment:
See comments below, but this seems like it would be simpler if, here in the app layer, we checked the right bits instead of having the store returned a count and going back to get the necessary data.

Reply:
I changed it so that the store returns an error if it fails to attach a file (ie. RowsAffected returns 0). The reason I don't want to do the extra GetForPost after attempting to attach the files is that failing to attach a post should be an extreme edge case, so I don't want to make an extra database hit on successful calls just because it might fail.